include/boost_plugin_loader/plugin_loader.hpp

```diff
@@ -95,21 +129,32 @@ PluginLoader::reportError(std::ostream& msg, const std::string& plugin_name, boo
   // Add information about the available plugins
   const std::string plugin_base_type = boost::core::demangle(typeid(PluginBase).name());
   auto plugins = getAvailablePlugins(PluginBase::getSection());
-  msg << "Available plugins of type '" << plugin_base_type << "':" << std::endl;
+  msg << "Available plugins of type '" << plugin_base_type << "':\n";
   for (const auto& p : plugins)
-    msg << "    - " + p << std::endl;
+    msg << "    - " + p << "\n";
 }
 
 template <class PluginBase>
 std::shared_ptr<PluginBase> PluginLoader::createInstance(const std::string& plugin_name) const
 {
   // Check for environment variable for plugin definitions
-  std::set<std::string> library_names = getAllLibraryNames(search_libraries_env, search_libraries);
-  if (library_names.empty())
+  const std::set<std::string> all_library_names = getAllLibraryNames(search_libraries_env, search_libraries);
+  if (all_library_names.empty())
     throw PluginLoaderException("No plugin libraries were provided!");
 
+  // Check for libraries provided as full paths. These are searched first
+  const std::array<std::set<std::string>, 2> separated_library_names =
+      separateLibraryPathSpecifications(all_library_names);
+  const std::set<std::string>& libraries_with_fullpath = separated_library_names[0];
+  const std::set<std::string>& library_names = separated_library_names[1];
```

Comment:
```suggestion
  std::set<std::string> libraries_with_fullpath;
  std::set<std::string> library_names;
  std::tie(libraries_with_fullpath, library_names) = separateLibraryPathSpecifications(all_library_names);
```

This would be much more clear and direct if the output of `separateLIbraryPathSpecifications` was a tuple and you could extract the outputs as shown in this comment

Reply:
I like the how it looks here so I will switch to a tuple.